tests/study_tests/test_study.py

```diff
@@ -1502,7 +1502,7 @@ def test_tell_from_another_process() -> None:
         assert study.best_value == 1.2
 
         # Should fail because the trial0 is already finished.
-        with pytest.raises(RuntimeError):
+        with pytest.raises(ValueError):
```

Comment:
FYI: Previously, `check_trial_is_updatable()` method in `optuna/storages/_rdb/storage.py` raises `RuntimeError`. However, from this change, `optuna/study/_tell.py"` raises `ValueError` like the following CI output.
https://github.com/optuna/optuna/actions/runs/4200669694/jobs/7286925478